pkg/sql/catalog/lease/storage.go

```diff
@@ -167,11 +169,15 @@ func (s storage) acquire(
 	for r := retry.StartWithCtx(ctx, retry.Options{}); r.Next(); {
 		err := s.db.Txn(ctx, acquireInTxn)
 		var pErr *roachpb.AmbiguousResultError
-		if errors.As(err, &pErr) {
+		switch {
+		case errors.As(err, &pErr):
 			log.Infof(ctx, "ambiguous error occurred during lease acquisition for %v, retrying: %v", id, err)
 			continue
-		}
-		if err != nil {
+		case pgerror.GetPGCode(err) == pgcode.UniqueViolation:
```

Comment:
Couldn't we instead salt the least significant bits of the jittered expiration timestamp somehow to prevent collisions?

Reply:
How does that help?

Reply:
The expiration timestamp is guaranteed to be unique for that node and since that's part of the PK we won't ever be inserting duplicate rows.

I'm obviously misunderstanding something here but I have no idea what.

Reply:
We're already jittering randomly by 5%. The hazard is that the same node picks a timestamp which is now + 5m +/- 5% and then releases it and then picks another one and it happens to be the same. This is a ~1/5M conflict (we store leases at microsecond resolution). Seems very very rare. Maybe there's something else going on, but I couldn't spot it.

Reply:
This matches my understanding. I was suggesting we maintain a global `var jitterCounter int64` with which we call `jitteredLeaseDuration(atomic.AddInt64(&jitterCounter, 1))` which salts the timestamp somehow, perhaps by overwriting the least significant 20 bits.

I took a bit more time to see how `rand.Float64` works and it seems exceedingly unlikely that collisions would occur unless `descriptor_lease_duration` and `descriptor_lease_jitter_fraction` were set to ridiculous values, which doesn't appear to be the case. The random number generator's internal state is protected by a mutex so each call to `rand.Float64` should generate a unique-enough value regardless of concurrency.

Reply:
So, the likelihood of a collision would be on the order or one in ten million: "rare but not rare enough" sounds about right to me.